jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -173,26 +206,120 @@ public void setIdleTimeout(Duration idleTimeout)
     public SocketAddress getBindAddress()
     {
         return bindAddress;
     }
 
+    /**
+     * <p>Sets the bind address of sockets before the connect operation.</p>
+     * <p>In multi-homed hosts, you may want to connect from a specific address:</p>
+     * <pre>
+     * clientConnector.setBindAddress(new InetSocketAddress("127.0.0.2", 0));
+     * </pre>
+     * <p>Note the use of the port {@code 0} to indicate that a different ephemeral port
+     * should be used for each different connection.</p>
+     * <p>In the rare cases where you want to use the same port for all connections,
+     * you must also call {@link #setReusePort(boolean) setReusePort(true)}.</p>
+     *
+     * @param bindAddress the socket address to bind to before the connect operation
+     */
     public void setBindAddress(SocketAddress bindAddress)
     {
         this.bindAddress = bindAddress;
     }
 
+    /**
+     * @return whether small TCP packets are sent without delay
+     */
+    @ManagedAttribute("Whether small TCP packets are sent without delay")
+    public boolean isTCPNoDelay()
+    {
+        return tcpNoDelay;
+    }
+
+    public void setTCPNoDelay(boolean tcpNoDelay)
+    {
+        this.tcpNoDelay = tcpNoDelay;
+    }
+
+    /**
+     * @return whether rebinding is allowed with sockets in tear-down states
+     */
+    @ManagedAttribute("Whether rebinding is allowed with sockets in tear-down states")
     public boolean getReuseAddress()
     {
         return reuseAddress;
     }
 
+    /**
+     * <p>Sets whether it is allowed to bind a socket to a socket address
+     * that may be in use by another socket in tear-down state, for example
+     * in TIME_WAIT state.</p>
+     * <p>This is useful when ClientConnector is restarted: an existing connection
+     * may still be using a network address (same host and same port) that is also
+     * chosen for a new connection.</p>
+     *
+     * @param reuseAddress whether rebinding is allowed with sockets in tear-down states
+     * @see #setReusePort(boolean)
+     */
     public void setReuseAddress(boolean reuseAddress)
     {
         this.reuseAddress = reuseAddress;
     }
 
+    /**
+     * @return whether binding to same host and port is allowed
+     */
+    @ManagedAttribute("Whether binding to same host and port is allowed")
+    public boolean isReusePort()
+    {
+        return reusePort;
+    }
+
+    /**
+     * <p>Sets whether it is allowed to bind multiple sockets to the same
+     * socket address (same host and same port).</p>
+     *
+     * @param reusePort whether binding to same host and port is allowed
+     */
+    public void setReusePort(boolean reusePort)
+    {
+        this.reusePort = reusePort;
+    }
+
+    /**
+     * @return the receive buffer size in bytes, or -1 for the default value
+     */
+    @ManagedAttribute("The receive buffer size in bytes")
+    public int getReceiveBufferSize()
```

Comment:
There appears no test for these new methods?

Reply:
Some property is tested, but some are trivially passed in as options.
Furthermore, some of these are just "hints" to the OS, so may or may not have effect.
I thought they were simple enough and deserved no testing.

Reply:
I was more looking for an example of the code that you just put into the javadoc at the start of this class (which has an error in it), but I see now that ClientConnector is already tested in a few places... but would be nice to see at least some tests call the setters for buffer size perhaps... not a blocker.